R/query.R

```diff
@@ -284,6 +284,22 @@ is_id_lookup <- function(expr) {
   expr$type == "lookup" && expr$name == "id"
 }
 
+is_logical <- function(expr) {
+  truthy <- c(quote(TRUE), quote(FALSE), quote(true), quote(false))
+  bool <- vlapply(truthy, function(x) x == expr)
+  sum(bool) == 1
+}
+
+as_logical <- function(expr) {
+  if (expr == quote(true)) {
+    ret <- quote(TRUE)
+  } else if (expr == quote(false)) {
+    ret <- quote(FALSE)
```

Comment:
```suggestion
    ret <- TRUE
  } else if (expr == quote(false)) {
    ret <- FALSE
```

no need to quote these, it  has no effect (`identical(TRUE, quote(TRUE))` is `TRUE`)